### DISCUSSION TITLE
Copilot cannot learn from custom function library's.

### DISCUSSION THREAD

**Discussion by nathangur** [community/community#18751] [Copilot Conversations]
I find it very irritating that when trying to get copilot to create something from a custom function library written in lua, it just gives me random gibberish instead of getting the library and repositories with that code from online. It doesn't seem smart, and you can't adjust it or feed it stuff.

**Comment by nathangur** [comment]
Is there any way to improve it or point it to the right direction? Like to a library or API website?

**Reply by dsetzer** [reply]
I really found it irritating too but but somewhat opposite of the problem you have.. for me copilot starts getting tunnel vision to the same code and functions and not providing anything new and not giving any alternatives to anything, sometimes just repeating the same line redundantly over and over again .

The only thing that would get it to actually provide suggestions using it's supposedly enormous codebase of all the code on github again was to make a new window with an empty folder for the project..

It really is very annoying that there's no setting to tell whether or not to make use of local code or exclude the local code.

I used to use tabnine which had settings to control this aspect but copilot seems to break whenever tabnine is active, seems something related to the inline suggestions.. would be nice to have them both working together..
